src/react/databrowser/objects/__tests__/ObjectUpload.test.jsx

```diff
@@ -0,0 +1,66 @@
+import * as s3object from '../../../actions/s3object';
+import { BUCKET_NAME } from '../../../actions/__tests__/utils/testUtil';
+import ObjectUpload from '../ObjectUpload';
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('ObjectUpload', () => {
+    it('should render ObjectUpload component', () => {
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(component.find(ObjectUpload).getDOMNode()).not.toBe(null);
+    });
+
+    it('should render an empty ObjectUpload component if showFolderCreate equals to false', () => {
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: false,
+            },
+        });
+
+        expect(component.find(ObjectUpload).getDOMNode()).toBe(null);
+    });
+
+    it('should call closeObjectUploadModal if cancel button is pressed', () => {
+        const closeObjectUploadModalMock = jest.spyOn(s3object, 'closeObjectUploadModal');
+
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(closeObjectUploadModalMock).toHaveBeenCalledTimes(0);
+        component.find('button#object-upload-cancel-button').simulate('click');
+        expect(closeObjectUploadModalMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not call uploadFiles if upload button is pressed when no files are present', () => {
+        const uploadFilesMock = jest.spyOn(s3object, 'uploadFiles');
+
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(uploadFilesMock).toHaveBeenCalledTimes(0);
+        component.find('button#object-upload-upload-button').simulate('click');
+        expect(uploadFilesMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should render NoFile component and should not render FileList when no files are present', () => {
+        const { component } = reduxMount(<ObjectUpload bucketName={BUCKET_NAME} prefixWithSlash=''/>, {
+            uiObjects: {
+                showObjectUpload: true,
+            },
+        });
+
+        expect(component.find('NoFile')).toHaveLength(1);
+        expect(component.find('FileList')).toHaveLength(0);
+    });
+});
```

Comment:
Can we also test 
- when one or multiple files are accepted?
- and then, when we press the upload and cancel button? 

`FileList` should be rendered and the related actions should be triggered.

Reply:
done: I found a way to change the initialValue in the ObjectUpload component so I can test if FileList is rendered

Reply:
It is good practice to write tests that are free of implementation details and continue to resemble the way our software is used as closely as possible.

I would recommend to use `.simulate('drop'` (instead of mocking the state) to populate the component's state.

Reply:
done